### PR TITLE
update: python-telethon

### LINKED ISSUE #3436: python-telethon: outdated

### PR DESCRIPTION
**Pull request by marianomarciello**
Update python-telethon from 1.13.0 to 1.24.0

Resolve  #3436

### CODE REVIEW

File: packages/python-telethon/PKGBUILD [hunk 1]
```diff
@@ -4,17 +4,17 @@
 pkgname=python-telethon
 _pkgname=Telethon
 _pyver=3.10
-pkgver=1.13.0
-pkgrel=4
+pkgver=1.24.0
+pkgrel=5
```
Comment (noraj) [inline packages/python-telethon/PKGBUILD:8]: set `pkgrel` back to `1` when upgrading `pkgver`

File: packages/python-telethon/PKGBUILD [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
 pkgdesc='Full-featured Telegram client library for Python 3.'
 arch=('any')
 url='https://pypi.org/project/telethon/#files'
 license=('MIT')
 depends=('python' 'python-pyaes' 'python-rsa')
 makedepends=('python-setuptools')
-source=("https://files.pythonhosted.org/packages/f4/44/547b3f020733724eaf202396f465957972dbaf6a4ef7aeff6fada6bcde28/$_pkgname-$pkgver.tar.gz")
-sha512sums=('5b039be72ede3abeb9a01513416b247af5cc4a37030add6dab2c78a070a88a145524e6f1e6b421b52721fc033e3fd39b331f3dc5933512b36e62598c82a5a42b')
+source=("https://files.pythonhosted.org/packages/71/75/40c3a381b230f7b5b4ea1b48df403590d54001a81e07cf819af812943822/$_pkgname-$pkgver.tar.gz")
+sha256sums=('818cb61281ed3f75ba4da9b68cb69486bed9474d2db4e0aa16e482053117452c')
```
Comment (noraj) [inline packages/python-telethon/PKGBUILD:17]: Use `sha512sums`, you can update with updpkgsum.

File: packages/python-telethon/PKGBUILD [hunk 3]
```diff
@@ -4,17 +4,17 @@
...
 epoch=1
 pkgdesc='Full-featured Telegram client library for Python 3.'
 arch=('any')
 url='https://pypi.org/project/telethon/#files'
 license=('MIT')
 depends=('python' 'python-pyaes' 'python-rsa')
 makedepends=('python-setuptools')
-source=("https://files.pythonhosted.org/packages/f4/44/547b3f020733724eaf202396f465957972dbaf6a4ef7aeff6fada6bcde28/$_pkgname-$pkgver.tar.gz")
-sha512sums=('5b039be72ede3abeb9a01513416b247af5cc4a37030add6dab2c78a070a88a145524e6f1e6b421b52721fc033e3fd39b331f3dc5933512b36e62598c82a5a42b')
+source=("https://files.pythonhosted.org/packages/71/75/40c3a381b230f7b5b4ea1b48df403590d54001a81e07cf819af812943822/$_pkgname-$pkgver.tar.gz")
```
Comment (noraj) [inline packages/python-telethon/PKGBUILD:16]: Use this for permaling https://github.com/BlackArch/blackarch-pkgbuilds/blob/4ae2d3f269ad76eda2d78f3ea172e9bba6fd1835/PKGBUILD-python-lib-setuptools#L15

Checking our templates is always a good idea ;)

**Review by noraj — CHANGES_REQUESTED** [review]

**Comment by noraj** [comment]
Have you check `genisys` still works with this dependency updated?

File: packages/python-telethon/PKGBUILD (hunk 3 above)
Comment (marianomarciello) [reply packages/python-telethon/PKGBUILD]: Hope my new commit will resolve this issue.

**Comment by marianomarciello** [comment]
> Have you check `genisys` still works with this dependency updated?

I do not use genisys, so I haven't done any test. I don't have any Telegram groups to manage.

**Comment by noraj** [comment]
> > Have you check `genisys` still works with this dependency updated?
> 
> I do not use genisys, so I haven't done any test. I don't have any Telegram groups to manage.

Ok because it's the only package in BA using this dependency.

**Comment by noraj** [comment]
I fixed the source URL 1636a960049b4c7a494d05856439ed9cd3ba1a87

Please use ba-dev to test your PR (https://gist.github.com/noraj/5045be32490f1161161e16fc5820066c), you would have spotted it:

```
$ ba-dev -b                                                                                                                                                                                                                                 
Syncing chroot copy /home/noraj/blackarch_chroot/blackarch/ with /home/noraj/blackarch_chroot/root/ ...                                                                                                                                      
[sudo] password for noraj:                                                                                                                                                                                                                   
Chroot environment is ready!                                                                                                                                                                                                                 
Consideer installing pkgcheck with pip install pkgcheck-arch --user for automatic PKGBUILD syntax error checking.                                                                                                                            
==> Making package: python-telethon 1:1.24.0-1 (dim. 10 juil. 2022 17:13:28)                                                                                                                                                                 
==> Retrieving sources...                                                                                                                                                                                                                    
  -> Downloading Telethon-1.24.0.tar.gz...                                                                                                                                                                                                   
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                                                                                                                                                              
                                 Dload  Upload   Total   Spent    Left  Speed                                                                                                                                                                
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0                                                                                                                                                               
curl: (22) The requested URL returned error: 404                                                                                                                                                                                             
==> ERROR: Failure while downloading https://files.pythonhosted.org/packages/source/T/Telethon-1.24.0.tar.gz                                                                                                                                 
    Aborting...                                                                                                                                                                                                                              
==> ERROR: Could not download sources.                                                                                                                                                                                                       
Failed to build the package.
```

**Comment by marianomarciello** [comment]
@noraj Thanks for your time and your patience even on a Sunday.

I am using currently makechrootpkg as suggested in [https://wiki.archlinux.org/title/DeveloperWiki:Building_in_a_clean_chroot#Setting_up_a_chroot](https://wiki.archlinux.org/title/DeveloperWiki:Building_in_a_clean_chroot#Setting_up_a_chroot).
I will start using ba-dev. Thanks

**Comment by noraj** [comment]
@marianomarciello You can use makechrootpkg too.

**Pull request merged**